app/presenters/person_presenter.rb

```diff
@@ -24,7 +24,7 @@ def full_hash_with_avatar
   def full_hash_with_profile
     attrs = full_hash
 
-    if own_profile? || person_is_following_current_user
+    if is_public || own_profile? || person_is_following_current_user
```

Comment:
My argument was to extract this into a method and also provide its result as attribute in the returned hash, and then use that attribute in the backbone view. That would de-duplicate and centralize the logic without having the client side guess depending on the returned fields.